test/Intl/GetCanonicalLocales.js

```diff
@@ -0,0 +1,142 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+var tests = [
+    {
+        name: "Intl.getCanonicalLocales Functionality (according to #sec-canonicalizelocalelist)",
+        // ensure array (or array-like) or convert to array, canonicalize each entry, remove duplicates
+        body: function () {
+            // ensure output is an array even if input was not an array
+            assert.areEqual(Intl.getCanonicalLocales('en'), ['en'], "Input is a singleton string (not an array) -> output is array");
+
+            // canonicalize case
+            assert.areEqual(Intl.getCanonicalLocales(['en']), ['en'], "Input matches output, no lookup is performed");
+            assert.areEqual(Intl.getCanonicalLocales(['en-us']), ['en-US'], "Canonicalize country casing (en-US) (all-lowercase)");
+            assert.areEqual(Intl.getCanonicalLocales(['en-Us']), ['en-US'], "Canonicalize country casing (en-US) (mixed-case)");
+            assert.areEqual(Intl.getCanonicalLocales(['EN-us']), ['en-US'], "Canonicalize country casing (en-US) (completely incorrect casing)");
+            assert.areEqual(Intl.getCanonicalLocales(['de-de']), ['de-DE'], "Canonicalize country casing (de-DE)");
+
+            // array-like objects are be fine (according to spec, arrays are converted to Object anyway)
+            // #sec-canonicalizelocalelist
+            //      5. Let len be ? ToLength(? Get(O, "length")).
+            //      6. Let k be 0.
+            //      7. Repeat, while k < len
+            // Since ToLength(undefined) === 0, we don't enter the loop (essentially treat the input as a zero-length array).
+            //      ToLength(undefined) -> ToInteger(undefined) -> ToNumber(undefined) -> NaN
+            //      ToInteger converts NaN to +0.
+            assert.areEqual(Intl.getCanonicalLocales({ '0': 'en-us' }), [], "Objects which might look like arrays are fine, but treated as 0 length.");
+            assert.areEqual(Intl.getCanonicalLocales({ 'a': 'b' }), [], "Arbitrary Objects are fine, treated as 0-length arrays.");
+
+            // TODO test this

```

Comment:
Yeah sorry about the TODOs. I wanted to get this out for review. I'll add more tests to cover the gaps in the next iteration.